olm.sh

```diff
@@ -77,6 +77,16 @@ for catalog in "${redhatCatalogs[@]}"; do
   yq -i eval 'del(.spec.install.spec.deployments[1].spec.template.spec.containers[0].securityContext.runAsGroup)' bundles/$catalog/$RELEASE/manifests/$package.clusterserviceversion.yaml
   yq -i eval 'del(.spec.install.spec.deployments[1].spec.template.spec.containers[0].securityContext.runAsUser)' bundles/$catalog/$RELEASE/manifests/$package.clusterserviceversion.yaml
 
+  # To provide channel for upgrade where we tell what versions can be replaced by the new version we offer
+  # You can read the documentation at link below:
+  # https://access.redhat.com/documentation/en-us/openshift_container_platform/4.2/html/operators/understanding-the-operator-lifecycle-manager-olm#olm-upgrades_olm-understanding-olm
+  echo "To provide channel for upgrading Operator..."
+  yq -i e '.metadata.annotations."olm.skipRange" |= "putversionrangehere"' bundles/$catalog/$RELEASE/manifests/$package.clusterserviceversion.yaml
+  rm -rf tmp.yaml
+  sed "s/putversionrangehere/'>=4.4.16 <${RELEASE}'/g" bundles/$catalog/$RELEASE/manifests/$package.clusterserviceversion.yaml > tmp.yaml
+  rm bundles/$catalog/$RELEASE/manifests/$package.clusterserviceversion.yaml
+  mv tmp.yaml bundles/$catalog/$RELEASE/manifests/$package.clusterserviceversion.yaml
```

Comment:
I feel this can be achieved in one inline replacement, right?

Reply:
Yes, I feel same, this can be better.
Thanks Shu, let me re-think the solution here!